shared-bindings/audiodelays/Echo.c

```diff
@@ -0,0 +1,300 @@
+// This file is part of the CircuitPython project: https://circuitpython.org
+//
+// SPDX-FileCopyrightText: Copyright (c) 2024 Mark Komus
+//
+// SPDX-License-Identifier: MIT
+
+#include <stdint.h>
+
+#include "shared-bindings/audiodelays/Echo.h"
+#include "shared-module/audiodelays/Echo.h"
+
+#include "shared/runtime/context_manager_helpers.h"
+#include "py/binary.h"
+#include "py/objproperty.h"
+#include "py/runtime.h"
+#include "shared-bindings/util.h"
+
+#define DECAY_DEFAULT 0.7f
+#define MIX_DEFAULT 0.5f
+
+//| class Echo:
+//|     """An Echo effect"""
+//|
+//|     def __init__(
+//|         self,
+//|         max_delay_ms: int = 500,
+//|         delay_ms: BlockInput = 250.0,
+//|         decay: BlockInput = 0.7,
+//|         mix: BlockInput = 0.5,
+//|         buffer_size: int = 512,
+//|         sample_rate: int = 8000,
+//|         bits_per_sample: int = 16,
+//|         samples_signed: bool = True,
+//|         channel_count: int = 1,
+//|     ) -> None:
+//|         """Create a Echo effect that echos an audio sample every set number of milliseconds.
+//|
+//|         :param int max_delay_ms: The maximum delay the echo can be
+//|         :param BlockInput delay_ms: The current echo delay
+//|         :param BlockInput decay: The rate the echo fades. 0.0 = instant; 1.0 = never.
+//|         :param BlockInput mix: The mix as a ratio of the sample (0.0) to the effect (1.0).
+//|         :param int buffer_size: The total size in bytes of the buffers to use
```

Comment:
I'd be clearer here that it only dictates the playback buffer, not the temporary storage of the echo.